crates/esthri/src/rusoto.rs

```diff
@@ -98,6 +106,52 @@ where
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum S3StorageClass {
+    Standard,
+    StandardIA,
+    IntelligentTiering,
+    OneZoneIA,
+    GlacialInstantRetrieval,
+    GlacialFlexibleRetrieval,
+    GlacialDeepArchive,
+    RRS,
+}
+
+impl From<S3StorageClass> for String {
+    fn from(s: S3StorageClass) -> Self {
+        match s {
+            S3StorageClass::Standard => "STANDARD",
+            S3StorageClass::StandardIA => "STANDARD_IA",
+            S3StorageClass::IntelligentTiering => "INTELLIGENT_TIERING",
+            S3StorageClass::OneZoneIA => "ONEZONE_IA",
+            S3StorageClass::GlacialInstantRetrieval => "GLACIER_IR",
+            S3StorageClass::GlacialFlexibleRetrieval => "GLACIER",
+            S3StorageClass::GlacialDeepArchive => "DEEP_ARCHIVE",
+            S3StorageClass::RRS => "REDUCED_REDUNDANCY",
+        }
+        .into()
+    }
+}
+
+impl TryFrom<&str> for S3StorageClass {
+    type Error = Error;
+
+    fn try_from(s: &str) -> Result<Self> {
+        match s {
+            "STANDARD" => Ok(S3StorageClass::Standard),
+            "STANDARD_IA" => Ok(S3StorageClass::StandardIA),
+            "INTELLIGENT_TIERING" => Ok(S3StorageClass::IntelligentTiering),
+            "ONEZONE_IA" => Ok(S3StorageClass::OneZoneIA),
+            "GLACIAL_INSTANT_RETRIEVAL" => Ok(S3StorageClass::GlacialInstantRetrieval),
+            "GLACIAL_FLEXIBLE_RETRIEVAL" => Ok(S3StorageClass::GlacialFlexibleRetrieval),
+            "GLACIAL_DEEP_ARCHIVE" => Ok(S3StorageClass::GlacialDeepArchive),
+            "REDUCED_REDUNDANCY" => Ok(S3StorageClass::RRS),
+            _ => Err(Error::UnknownStorageClass(s.to_string())),
+        }
+    }
+}
```

Comment:
This should probably be a [FromStr](https://doc.rust-lang.org/std/str/trait.FromStr.html) implementation. `TryFrom<&str>` is semantically the same thing, but it's more idiomatic to go from str -> a type via `FromStr`